packages/turf-clean-coords/index.js

```diff
@@ -0,0 +1,109 @@
+var clone = require('@turf/clone');
+var invariant = require('@turf/invariant');
+var getCoords = invariant.getCoords;
+var getGeomType = invariant.getGeomType;
+
+/**
+ * Removes redundant coordinates from a (Multi)LineString or (Multi)Polygon; ignores (Multi)Point.
+ *
+ * @name cleanCoords
+ * @param {Geometry|Feature<any>} geojson Feature or Geometry
+ * @param {boolean} [mutate=false] allows GeoJSON input to be mutated
+ * @returns {Geometry|Feature<any>} the cleaned input Feature/Geometry
+ * @example
+ * var line = trf.lineString([[0, 0], [0, 2], [0, 5], [0, 8], [0, 8], [0, 10]]);
+ *
+ * var cleaned = turf.cleanCoords(line).geometry.coordinates;
+ * //= [[0, 0], [0, 10]]
+ */
+module.exports = function (geojson, mutate) {
+    if (!geojson) throw new Error('geojson is required');
+    var type = getGeomType(geojson);
+    var newPoints = [];
+
+    var coords = getCoords(geojson);
+
+    switch (type) {
+    case 'LineString':
+        newPoints = cleanCoords(geojson, mutate);
+        break;
+    case 'MultiLineString':
+    case 'Polygon':
+        for (var i = 0; i < coords.length; i++) {
+            var line = coords[i];
+            newPoints.push(cleanCoords(line));
+        }
+        break;
+    case 'MultiPolygon':
+        for (var j = 0; j < coords.length; j++) {
+            var polys = coords[j];
+            var polyPoints = [];
+            for (var p = 0; p < polys.length; p++) {
+                var ring = polys[p];
+                polyPoints.push(cleanCoords(ring));
+            }
+            newPoints.push(polyPoints);
+        }
+        break;
+    case 'Point':
+    case 'MultiPoint':
+        newPoints = coords;
+        break;
+    default:
+        throw new Error(type + ' geometry not supported');
+    }
+
+    var output = (mutate === true) ? geojson : clone(geojson, true);
```

Comment:
2nd - Since we aren't mutating `properties` we don't need to define `clone(geojson, true)` as TRUE, if we use the default or `FALSE` it will only clone the coordinates (which is a lot faster process).